File: src/Algorithms/Greedy/easy/MinimumAbsoluteDifferenceInAnArray.java
//Date 25-9-2018

package Algorithms.Greedy.easy;

import java.util.Arrays;
import java.util.Scanner;

public class MinimumAbsoluteDifferenceInAnArray {

    public static void main(String[] args){
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int [] arr = new int [n];
        for(int i=0;i<n;i++)
            arr[i]=in.nextInt();

        Arrays.sort(arr);
        int min = Math.abs(arr[0]- arr[1]);
        int current;
        for(int i=1;i<n-1;i++){
            current = Math.abs(arr[i]-arr[i+1]);
            min = ( min > current) ? current : min ;
            if( min == 0)
                break;
        }
        System.out.println(min);
    }

}

File: src/TheHackerRankInterviewPreparationKit/Greedy_Algorithms/Minimum Absolute Difference in an Array.kt
package TheHackerRankInterviewPreparationKit.Greedy_Algorithms
//Date 12-8-2018
import java.util.*

fun main(args : Array<String>){

    val s = Scanner(System.`in`)
    val n = s.nextInt()
    val arr = IntArray(n)

    for( i in 0 until n)
        arr[i] = s.nextInt()

    Arrays.sort(arr)

    var min = Math.abs(arr[0]-arr[1])
    var current: Int

    for( i in 0 until n-1)
    {
        current = Math.abs(arr[i]-arr[i+1])
        if( current < min)
        {
            min = current
            if( min == 0)
                break
        }
    }
    println(min)
    s.close()
}